File: app/src/main/java/com/omvp/app/ui/samples/requestphone/presenter/SampleRequestPhonePresenterImpl.java

package com.omvp.app.ui.samples.requestphone.presenter;


import android.text.TextUtils;

import com.omvp.app.R;
import com.omvp.app.base.mvp.presenter.BasePresenter;
import com.omvp.app.interceptor.authPhone.AuthPhoneInterceptor;
import com.omvp.app.ui.samples.requestphone.view.SampleRequestPhoneView;

import javax.inject.Inject;

public class SampleRequestPhonePresenterImpl extends BasePresenter<SampleRequestPhoneView> implements SampleRequestPhonePresenter {

    @Inject
    AuthPhoneInterceptor mAuthPhoneInterceptor;

    @Inject
    public SampleRequestPhonePresenterImpl(SampleRequestPhoneView sampleRequestPhoneFlowView) {
        super(sampleRequestPhoneFlowView);
    }

    @Override
    public void onViewLoaded() {
        super.onViewLoaded();

    }

    @Override
    public void validatePhone(String prefix, String phone) {
        boolean formValid = true;
        if (TextUtils.isEmpty(prefix)) {
            showPrefixInputError(mContext.getString(R.string.invalid_prefix));
            formValid = false;
        } else {
            showPrefixInputSuccess();
        }
        if (TextUtils.isEmpty(phone)) {
            showPhoneInputError(mContext.getString(R.string.enter_your_phone));
            formValid = false;
        } else {
            showPhoneInputSuccess();
        }
        if (formValid) {
            startPhoneNumberVerification("+" + prefix + phone);
        }
    }

    private void startPhoneNumberVerification(String number) {
        showProgress();
        mAuthPhoneInterceptor.startPhoneNumberVerification(number);
    }

    private void showPrefixInputError(String error) {
        if (getMView() != null) {
            getMView().showPrefixInputError(error);
        }
    }

    private void showPhoneInputError(String error) {
        if (getMView() != null) {
            getMView().showPhoneInputError(error);
        }
    }

    private void showPrefixInputSuccess() {
        if (getMView() != null) {
            getMView().showPrefixInputSuccess();
        }
    }

    private void showPhoneInputSuccess() {
        if (getMView() != null) {
            getMView().showPhoneInputSuccess();
        }
    }
}

File: app/src/main/java/com/omvp/app/ui/samples/requestphone/presenter/SampleRequestPhonePresenterImpl.kt
package com.omvp.app.ui.samples.requestphone.presenter

import com.omvp.app.R
import com.omvp.app.base.mvp.presenter.BasePresenter
import com.omvp.app.interceptor.authPhone.AuthPhoneInterceptor
import com.omvp.app.ui.samples.requestphone.view.SampleRequestPhoneView
import javax.inject.Inject

class SampleRequestPhonePresenterImpl
@Inject
constructor(sampleRequestPhoneView: SampleRequestPhoneView) : BasePresenter<SampleRequestPhoneView>(sampleRequestPhoneView), SampleRequestPhonePresenter {

    @Inject
    internal lateinit var mAuthPhoneInterceptor: AuthPhoneInterceptor

    override fun validatePhone(prefix: String, phone: String) {
        var formValid = true
        if (prefix.isEmpty()) {
            showPrefixInputError(mContext.getString(R.string.invalid_prefix))
            formValid = false
        } else run { showPrefixInputSuccess() }

        if (phone.isEmpty()) {
            showPhoneInputError(mContext.getString(R.string.enter_your_phone))
            formValid = false
        } else run { showPhoneInputSuccess() }
        if (formValid) {
            startPhoneNumberVerification("+$prefix$phone")
        }
    }

    private fun startPhoneNumberVerification(number: String) {
        showProgress()
        mAuthPhoneInterceptor.startPhoneNumberVerification(number)
    }

    private fun showPrefixInputError(error: String) {
        mView?.showPrefixInputError(error)
    }

    private fun showPhoneInputError(error: String) {
        mView?.showPhoneInputError(error)
    }

    private fun showPrefixInputSuccess() {
        mView?.showPrefixInputSuccess()
    }

    private fun showPhoneInputSuccess() {
        mView?.showPhoneInputSuccess()
    }
}
